core/templates/components/ck-editor-helpers/ck-editor-copy-content-service.spec.ts

```diff
@@ -0,0 +1,200 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the Ck editor copy content service.
+ */
+
+import { CkEditorCopyContentService } from
+  'components/ck-editor-helpers/ck-editor-copy-content-service';
+import { HtmlEscaperService } from 'services/html-escaper.service';
+import { LoggerService } from 'services/contextual/logger.service';
+
+/**
+ * Returns a HTMLElement containing the element to copy.
+ * @param {string} html The hmtl to place in angular-html-bind element.
+ * @returns {HTMLElement} The element created by param html.
+ */
+const generateContent = (html: string): HTMLElement => {
+  const container = document.createElement('template');
+  container.innerHTML = `<angular-html-bind>${html.trim()}</angular-html-bind>`;
+  return <HTMLElement>(container.content.firstChild.firstChild);
+};
+
+
+describe('Ck editor copy content service', () => {
+  let loggerService = new LoggerService();
+  let htmlEscaperService = new HtmlEscaperService(loggerService);
+
+  let service: CkEditorCopyContentService;
+  let ckEditorStub: Partial<CKEDITOR.editor>;
+
+  beforeEach(() => {
+    ckEditorStub = {
+      id: 'editor1',
+      insertHtml: (_: string) => {},
+      execCommand: (_: string): boolean => true,
+    };
+    service = new CkEditorCopyContentService(htmlEscaperService);
+  });
+
+  it('not broadcast copy if copy mode is not active', () => {
+    expect(service.copyModeActive).toBe(false);
+
+    const pElement = generateContent('<p>Hello</p>');
+    const insertHtmlSpy = spyOn(ckEditorStub, 'insertHtml');
+    const execCommandSpy = spyOn(ckEditorStub, 'execCommand');
+
+    service.bindPasteHandler(ckEditorStub);
+    service.broadcastCopy(pElement);
+
+    expect(insertHtmlSpy).not.toHaveBeenCalled();
+    expect(execCommandSpy).not.toHaveBeenCalled();
```

Comment:
This assert's that the copy is not done, right? but the test is for broadcast doesn't happen, should we (firstly) assert that the `copyEventEmitter` not to have called? 

[Also, should we change the `it`message to `brodcastCopy should not emit copy event if copy mode dsiabled` (or something around this]

You can follow this pattern for the test if you want:
```
describe: ckEcitor copy servi
    describe: broadcastCopy
        it: should not emit ...
        it: should emit copy ...

```
(not sure whether this will be helpful!)

Reply:
Given that ```copyEventEmitter``` is private, I think it is fine to not check that it has been called. I think this test still correctly tests behavior, rather than the implementation.